core/src/main/java/org/kiwix/kiwixmobile/core/error/ErrorActivity.kt

```diff
@@ -0,0 +1,230 @@
+import kotlinx.android.synthetic.main.activity_kiwix_error.allowDeviceDetails
+import kotlinx.android.synthetic.main.activity_kiwix_error.allowFileSystemDetails
+import kotlinx.android.synthetic.main.activity_kiwix_error.allowLanguage
+import kotlinx.android.synthetic.main.activity_kiwix_error.allowLogs
+import kotlinx.android.synthetic.main.activity_kiwix_error.allowZims
+import kotlinx.android.synthetic.main.activity_kiwix_error.reportButton
+import kotlinx.android.synthetic.main.activity_kiwix_error.restartButton
+import org.kiwix.kiwixmobile.core.R
+import org.kiwix.kiwixmobile.core.base.BaseActivity
+import org.kiwix.kiwixmobile.core.dao.NewBookDao
+import org.kiwix.kiwixmobile.core.di.components.CoreComponent
+import org.kiwix.kiwixmobile.core.reader.ZimReaderContainer
+import org.kiwix.kiwixmobile.core.utils.LanguageUtils.Companion.getCurrentLocale
+import org.kiwix.kiwixmobile.core.utils.files.FileLogger
+import org.kiwix.kiwixmobile.zim_manager.MountPointProducer
+import java.io.PrintWriter
+import java.io.StringWriter
+import javax.inject.Inject
+import kotlin.system.exitProcess
+
+private const val STATUS = 10
+private const val ZERO = 0
+
+open class ErrorActivity : BaseActivity() {
+  @Inject
+  lateinit var bookDao: NewBookDao
+
+  @Inject
+  lateinit var zimReaderContainer: ZimReaderContainer
+
+  @Inject
+  lateinit var mountPointProducer: MountPointProducer
+
+  @Inject
+  lateinit var fileLogger: FileLogger
+
+  private var exception: Throwable? = null
+  private lateinit var emailIntent: Intent
+  private val deviceDetails = """Device Details:
+        Device:[${Build.DEVICE}]
+        Model:[${Build.MODEL}]
+        Manufacturer:[${Build.MANUFACTURER}]
+        Time:[${Build.TIME}]
+        Android Version:[${Build.VERSION.RELEASE}]
+        App Version:[$versionName $versionCode]
+        
+        """
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_kiwix_error)
+    val callingIntent = intent
+    val extras = callingIntent.extras
+    exception = if (extras != null && safeContains(extras)) {
+      extras.getSerializable(EXCEPTION_KEY) as Throwable
+    } else {
+      null
+    }
+    addExtrasToEmailIntent()
+    setupReportButton()
+    restartButton.setOnClickListener { restartApp() }
+  }
+
+  private fun setupReportButton() {
+    reportButton.setOnClickListener {
+      startActivityForResult(
+        Intent.createChooser(emailIntent, "Send email..."), 1
+      )
+    }
+  }
+
+  private fun addExtrasToEmailIntent() {
+    emailIntent = Intent(Intent.ACTION_SEND)
+    with(emailIntent) {
+      type = "vnd.android.cursor.dir/email"
+      putExtra(
+        Intent.EXTRA_EMAIL,
+        arrayOf("android-crash-feedback@kiwix.org")
+      )
+      putExtra(Intent.EXTRA_SUBJECT, subject)
+      putExtra(Intent.EXTRA_TEXT, buildBody())
+    }
+    if (allowLogs.isChecked) {
```

Comment:
this if can be inside the `apply/with` to qualify the `this` you access you can use `this@ErrorActivity`